### ISSUE TITLE
Support Deployment of SR-IOV Network Operator via helm

### ISSUE THREAD

**Issue by adrianchiris** [k8snetworkplumbingwg/sriov-network-operator#56]
This Enhancement proposes to add [Helm ](https://helm.sh/)charts to the project to allow an easy depoyment of sriov-network-operator in a k8s environment.

The charts will be added in a new `deployment` folder under the project root.

upon release the chart parameter can be modified to point to a release version of sriov network operator images and possibly be made available in a [helm repository](https://helm.sh/docs/topics/chart_repository/). 
one way of doing that is using [github pages](https://pages.github.com/) to host the repository (index file, packaged charts and possibly some project documentation)

However this can be regarded as a followup effort.

**Comment by e0ne** [comment]
Helm chars is added in #76. We still need to create helm chart repository to allow people use charts from it

**Comment by adrianchiris** [comment]
Discussed the Helm Repository part of this issue in today's community meeting.

having a helm repository is desirable and github pages seems like a viable option to do it.
@martinkennelly is also +1 on it.

@zshi-redhat @pliurh do you have any inputs / preferences / objections ?

At the end the user would have to do the following to deploy sriov-network-operator:
```
$ helm repo sno-repo https://k8snetworkplumbingwg.github.io/sriov-network-operator
$ helm repo update
$ helm install -n sriov-network-operator --create-namespace --wait sriov-network-operator sno-repo/sriov-network-operator
```

An alternative could be to have a place for the entire org to host project charts (A gh_pages branch in https://github.com/k8snetworkplumbingwg/helm-charts ?)
This would need to be raised in the npwg bi-weekly meeting.

A downside for this approach is that it would not be self contained.

**Comment by pliurh** [comment]
> An alternative could be to have a place for the entire org to host project charts (A gh_pages branch in https://github.com/k8snetworkplumbingwg/helm-charts ?)

I prefer this approach. @dougbtv What's your suggestion?

**Comment by ianb-mp** [comment]
It'd be great to see a solution to this. A central repo for helm-charts sounds like the right approach, unfortunately it looks like https://github.com/k8snetworkplumbingwg/helm-charts/ has been abandoned

**Comment by SchSeba** [comment]
That is fine we can push on having just a github pages to support it

**Comment by ianb-mp** [comment]
> That is fine we can push on having just a github pages to support it

@SchSeba I had a look at this. A common pattern for publishing helm charts is to use the [Chart Releaser](https://github.com/helm/chart-releaser-action) Github action to update `index.yml` in `gh-pages` branch, and push a release & chart tar file to the main/master branch. Alternatively, the chart tar can be stored in `gh-pages` branch. Before this can be setup, I think the release process needs to be clarified. For example, it looks like releases are currently done manually and not driven from Actions.

**Comment by ianb-mp** [comment]
> Agree that are stuffs we need to talk about will it be possible for you to join our next community meeting?

@SchSeba I would like to attend, however that meeting is at midnight in my timezone which makes it difficult for me. BTW, did you mean to post the doc edit link?

**Comment by SchSeba** [comment]
sure the link is open to everyone who wants to contribute :) 

I am sorry that the time zone doesn't work for you.
do you want to open a PR and we can review the PR?

**Comment by ianb-mp** [comment]
> do you want to open a PR and we can review the PR?

@SchSeba to be clear, the PR would be to host helm chart in sriov-network-operator repo (and not use https://github.com/k8snetworkplumbingwg/helm-charts/ repo)? 

@dougbtv @e0ne do either of you have thoughts on this? If the 'helm-charts' repo is no longer to be used, it should be archived.

**Comment by SchSeba** [comment]
I am fine to host the sriov-operator helm in this repo

**Comment by ianb-mp** [comment]
It looks this is fixed by #715 

GHCR hosts the helm chart which can be used like this:

```sh
$ helm install -n sriov-network-operator --create-namespace \
  --version 1.3.0 \
  --set sriovOperatorConfig.deploy=true sriov-network-operator \
  oci://ghcr.io/k8snetworkplumbingwg/sriov-network-operator-chart
```
